packages/core/src/contrib/Functions.ts

```diff
@@ -1550,6 +1549,106 @@ export const compDict = {
   unitVector: (_context: Context, theta: ad.Num): VectorV<ad.Num> => {
     return { tag: "VectorV", contents: [cos(theta), sin(theta)] };
   },
+
+  /**
+   * Helper function for closestPoint to calculate closest point of rectangle by comparing
+   * min abs vals between point vs edge of rectangle
+   */
+  // minComp: (
+  //   _context: Context,
+  //   p1: ad.Num[], //Rectangle corner in 1st quadrant
+  //   p2: ad.Num[], //Point coordinates
+  //   center: ad.Num[]
+  // ): VectorV<ad.Num> => {
+  //   p2 = ops.vsub(p2, center);
+  //   const X = absVal(p1[0]) < absVal(p2[0]) ? p1[0] : p2[0];
+  //   const Y = absVal(p1[1]) < absVal(p2[1]) ? p1[1] : p2[1];
+  //   return { tag: "VectorV", contents: ops.vadd([X, Y], center) };
+  // },
+//
+  closestPoint: (
+    _context: Context,
+    [t, s]: [string, any],
+    p: ad.Num[]
+  ): VectorV<ad.Num> => {
+    /**
+     * Helper function for Rectangles
+     * @param _context
+     * @param x
+     * @param a
+     * @param b
+     * @returns
+     */
+    const clamp = (
+      _context: Context,
+      x: ad.Num,
+      l: ad.Num,
+      u: ad.Num
+    ): ad.Num => {
+      return max(l, min(u, x));
+    };
+    const getNearestPointRect = (
+      _context: Context,
+      l: ad.Num,
+      t: ad.Num,
+      w: ad.Num,
+      h: ad.Num,
+      x: ad.Num,
+      y: ad.Num
+    ): ad.Num[] => {
+      const r = add(l, w);
+      const b = add(t, h); //Formatted using JavaGraphics coordinate system
+      console.log(r);
+      console.log(b);
+      x = clamp(_context, x, l, r);
```

Comment:
```suggestion
      x = clamp(x, l, r);
```
Can remove context argument if no longer needed in clamp's function body.